src/main/scala/com/atomist/tree/marshal/LinkedJsonTreeDeserializer.scala

```diff
@@ -103,6 +102,21 @@ object LinkedJsonTreeDeserializer extends LazyLogging {
     }
 }
 
+class EmptyLinkableContainerTreeNode() extends ContainerTreeNode {
+
+  override def nodeName: String = "empty"
+
+  override def value: String = ""
```

Comment:
This one is more questionable. Null is probably more correct. However I went with "" because I want this node to be valid. It does get processed by the path expression, it's just a TreeNode that should never match anything. Similar to having empty sets / lists I didn't want to run this risk of causing a NPE somewhere down the line.

That said I do think it's questionable. What do you think?